Dockerfile

```diff
@@ -20,87 +20,146 @@
-COPY $EXTERNAL_SOURCE/packages/backend/package.json ./packages/backend/package.json
+COPY $EXTERNAL_SOURCE_NESTED/package.json $EXTERNAL_SOURCE_NESTED/yarn.lock ./
+COPY $EXTERNAL_SOURCE_NESTED/packages/app/package.json ./packages/app/package.json
+COPY $EXTERNAL_SOURCE_NESTED/packages/backend/package.json ./packages/backend/package.json
 
 # see https://docs.engineering.redhat.com/pages/viewpage.action?pageId=228017926#UpstreamSources(Cachito,ContainerFirst)-CachitoIntegrationfornpm
 # Downstream uncomment
-# RUN source $CONTAINER_SOURCE/cachito.env && $YARN config set nodedir /usr; $YARN config set unsafe-perm true; # $YARN config list --verbose
+# COPY $REMOTE_SOURCES/upstream1/cachito.env \
+#      $REMOTE_SOURCES/upstream1/app/registry-ca.pem \
+#      $REMOTE_SOURCES/upstream1/app/distgit/containers/rhdh-hub/.npmrc \
+#      ./
+# # registry=https://cachito-nexus.engineering.redhat.com/repository/cachito-yarn-914335/
+# # email=noreply@domain.local
+# # always-auth=true
+# # //cachito-nexus.engineering.redhat.com/repository/cachito-yarn-914335/:_auth=Y2FjaGl0by15YXJuLTkxNDMzNTo2OWZjM2Q1NDVhMTA5ZGU1OGJmM2E5ZTliYg==
+# # fetch-retries=5
+# # fetch-retry-factor=2
+# # strict-ssl=true
+# # cafile="../../../registry-ca.pem"
+# # hadolint ignore=SC1091
+# RUN \
+#     # debug
+#     # ls -l $CONTAINER_SOURCE/cachito.env; \ 
+#     # load envs
+#     source $CONTAINER_SOURCE/cachito.env; \
+#     \
+#     # load cert
+#     cert_path=$CONTAINER_SOURCE/registry-ca.pem; \
+#     # debug
+#     # ls -la "${cert_path}"; \
+#     npm config set cafile "${cert_path}"; $YARN config set cafile "${cert_path}" -g; \
+#     \
+#     # debug
+#     # ls -l /usr/; \
+#     # set up node dir with common.gypi and unsafe-perms=true
+#     ln -s /usr/include/node/common.gypi /usr/common.gypi; $YARN config set nodedir /usr; $YARN config set unsafe-perm true
+#     # debug
+#     # cat $CONTAINER_SOURCE/.npmrc || true; \
+#     # $YARN config list --verbose; npm config list; npm config list -l
 #/ Downstream uncomment
 
 RUN $YARN install --frozen-lockfile --network-timeout 600000
 
 # Stage 3 - Build packages
+# Downstream comment
 FROM deps AS build
+#/ Downstream comment
 
-COPY $EXTERNAL_SOURCE ./
+COPY $EXTERNAL_SOURCE_NESTED ./
 
 RUN git config --global --add safe.directory ./
-RUN rm app-config.yaml && mv app-config.example.yaml app-config.yaml
-
-# see https://docs.engineering.redhat.com/pages/viewpage.action?pageId=228017926#UpstreamSources(Cachito,ContainerFirst)-CachitoIntegrationfornpm
-# Downstream uncomment
-# RUN source $CONTAINER_SOURCE/cachito.env && $YARN config set nodedir /usr; $YARN config set unsafe-perm true; # $YARN config list --verbose
-#/ Downstream uncomment
+# Downstream comment
+# RUN rm app-config.yaml && mv app-config.example.yaml app-config.yaml
+#/ Downstream comment
 
+# hadolint ignore=DL3059
 RUN $YARN build --filter=backend
 
 # Stage 4 - Build the actual backend image and install production dependencies
+# Downstream comment
```

Comment:
if you prefer, I can just skip these unsupported intermediate stages with a sed command -- https://gitlab.cee.redhat.com/rhidp/cpaas-rhdh-hub/-/blob/rhdh-1.0-rhel-9/sync.sh#L242-243